airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/run.py

```diff
@@ -4,13 +4,51 @@
 
 
 import sys
+import time
+import traceback
+from typing import List
 
-from airbyte_cdk.entrypoint import launch
+from airbyte_cdk.entrypoint import AirbyteEntrypoint, launch
+from airbyte_cdk.models import AirbyteErrorTraceMessage, AirbyteMessage, AirbyteMessageSerializer, AirbyteTraceMessage, TraceType, Type
+from orjson import orjson
 from source_amazon_ads import SourceAmazonAds
 from source_amazon_ads.config_migrations import MigrateStartDate
 
 
+def _get_source(args: List[str]):
+    catalog_path = AirbyteEntrypoint.extract_catalog(args)
+    config_path = AirbyteEntrypoint.extract_config(args)
+    state_path = AirbyteEntrypoint.extract_state(args)
+    try:
+        return SourceAmazonAds(
+            SourceAmazonAds.read_catalog(catalog_path) if catalog_path else None,
+            SourceAmazonAds.read_config(config_path) if config_path else None,
+            SourceAmazonAds.read_state(state_path) if state_path else None,
+        )
+    except Exception as error:
+        print(
+            orjson.dumps(
+                AirbyteMessageSerializer.dump(
+                    AirbyteMessage(
+                        type=Type.TRACE,
+                        trace=AirbyteTraceMessage(
+                            type=TraceType.ERROR,
+                            emitted_at=time.time_ns() // 1_000_000,
+                            error=AirbyteErrorTraceMessage(
+                                message=f"Error starting the sync. This could be due to an invalid configuration or catalog. Please contact Support for assistance. Error: {error}",
+                                stack_trace=traceback.format_exc(),
+                            ),
+                        ),
+                    )
+                )
+            ).decode()
+        )
+        return None
```

Comment:
We should probably:
* `init_uncaught_exception_handler`
* raise here

Else, the error message won't be so clear if the SourceAmazonAds.__init__ fails

Reply:
added `init_uncaught_exception_handler`